tests/test_project_time.py

```diff
@@ -0,0 +1,251 @@
+    """
+    ptime_col = project_time_dim.get_timestamp_load_data_columns()
+    assert len(ptime_col) == 1, ptime_col
+    ptime_col = ptime_col[0]
+
+    time_cols = tempo._handler.config.get_dimension(
+        DimensionType.TIME
+    ).get_timestamp_load_data_columns()
+    enduse_cols = tempo._handler.get_pivoted_dimension_columns()
+
+    # get sum from converted_data
+    groupby_cols = [col for col in converted_data.columns if col not in enduse_cols + [ptime_col]]
+    converted_sum = converted_data.groupBy(*groupby_cols).agg(
+        *[F.sum(col).alias(col) for col in enduse_cols]
+    )
+    converted_sum_df = converted_sum.toPandas().set_index(groupby_cols).sort_index()
+
+    # process raw_data, get freq each values will be mapped and get sumproduct from there
+    session_tz = _get_spark_session().conf.get("spark.sql.session.timeZone")
+    model_time = (
+        pd.Series(np.concatenate(project_time_dim.list_expected_dataset_timestamps()))
+        .rename(ptime_col)
+        .to_frame()
+    )
+    model_time[ptime_col] = model_time[ptime_col].dt.tz_convert(session_tz)
+
+    geo_tz_values = [row.time_zone for row in raw_data.select("time_zone").distinct().collect()]
+    geo_tz_names = [get_timezone(tz).tz_name for tz in geo_tz_values]
+
+    model_time_df = []
+    for tzv, tz in zip(geo_tz_values, geo_tz_names):
+        model_time_tz = model_time.copy()
+        model_time_tz["time_zone"] = tzv
+        model_time_tz["new_time"] = model_time_tz[ptime_col].dt.tz_convert(tz)  # <---
+        for col in time_cols:
+            if col == "hour":
+                model_time_tz[col] = model_time_tz[ptime_col].dt.tz_convert(tz).dt.hour
+            elif col == "day_of_week":
+                model_time_tz[col] = model_time_tz[ptime_col].dt.tz_convert(tz).dt.day_of_week
+            elif col == "month":
+                model_time_tz[col] = model_time_tz[ptime_col].dt.tz_convert(tz).dt.month
+            else:
+                raise ValueError(f"{col} does not have a function specified in test.")
+        model_time_df.append(model_time_tz)
+
+    model_time_df = pd.concat(model_time_df, axis=0)
+    model_time_df = (
+        model_time_df.groupby(["time_zone"] + time_cols)[ptime_col]
+        .count()
+        .rename("count")
+        .to_frame()
+    )
+    raw_data_df = raw_data.toPandas().join(
+        model_time_df, on=["time_zone", "month", "day_of_week", "hour"], how="left"
+    )
+
+    # check that the sum of frequency count is 8784
+    n_ts = raw_data_df.groupby(groupby_cols)["count"].sum().unique()
+    assert list(n_ts) == [
+        len(model_time)
+    ], f"Mismatch in number of timestamps: {n_ts} vs. {len(model_time)}"
+
+    raw_data_df[enduse_cols] = raw_data_df[enduse_cols].multiply(raw_data_df["count"], axis=0)
+    raw_sum_df = raw_data_df.groupby(groupby_cols)[enduse_cols].sum().sort_index()
+
+    # compare annual sums
+    delta_df = converted_sum_df - raw_sum_df
```

Comment:
Is there any chance of minor differences in these numbers?

Reply:
Certainly reasonable to use np.isclose or math.isclose here

Reply:
Do not need to round here because the load data used in either method came from tempo load_data spark df converted toPandas(), so they should retain the same level of precision.